apps/public/public.go

```diff
@@ -372,7 +372,11 @@ func (pca Client) browserLogin(ctx context.Context, redirectURI *url.URL, params
 		return interactiveAuthResult{}, err
 	}
 	defer srv.Shutdown()
-	authURL, err := pca.base.AuthCodeURL(ctx, params.ClientID, srv.Addr, params.Scopes, params)
+	replyUrl := srv.Addr
+	if redirectURI != nil {
+		replyUrl = redirectURI.String()
```

Comment:
For interactive auth, the redirect URI is always going to be `http://localhost` (it's currently the loop-back address which we need to change).

The API takes a URI parameter as a means to specify a custom port number only (kinda clunky IMO but it's how the MSALs work).

Reply:
I'd be perfectly fine with localhost as that would work as a callback to
the well-known CLI application ID.

As a side note, are there any plans to have a cleaner integration with the
azure-autorest SDK's? Currently I'm using the return from this and then
doing a few type conversions to make it work over there.  I know this is
all pre-beta, but Go is the language of choice for all our internal
platform tooling (large enterprise customer).

Reply:
The short answer is yes.

The long answer.  We have a new authentication package, [`azidentity`](https://github.com/Azure/azure-sdk-for-go/tree/master/sdk/azidentity) to be used to authenticate with our "track 2" (i.e. next-gen) Azure SDK for Go (the content under the [`./sdk`](https://github.com/Azure/azure-sdk-for-go/tree/master/sdk) directory contains the track 2 packages).  There's also an adapter, [`azidext`](https://github.com/jongio/azidext/tree/master/go/azidext), that you can use to adapt `azidentity` authentication with the track 1 SDK packages.  Once we have a preview of this MSAL package, we will retool the innards of `azidentity` to consume it.

Reply:
Fantastic -- thank you for the insight, @jhendrixMSFT